packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -433,6 +434,27 @@ void main() {
     expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 0.5);
   });
 
+  testWidgets('Switch is using track color when set', (WidgetTester tester) async {
+    const Color trackColor = Colors.pink;
+
+    await tester.pumpWidget(
+      const Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: CupertinoSwitch(
+            value: false,
+            trackColor: trackColor,
+            dragStartBehavior: DragStartBehavior.down,
+            onChanged: null,
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(CupertinoSwitch), findsOneWidget);
+    expect(tester.widget<CupertinoSwitch>(find.byType(CupertinoSwitch).first).trackColor, trackColor);
```

Comment:
Optional: maybe add a few lines here to verify that if you change `trackColor` the switch on screen will be repainted in a different color.

Reply:
Yeah, I think because constant nature of `Color` this is governed by a widget above the tree. So eventually this should be rendered anyway.